From: Michael Lee <michael-cy.lee@mediatek.com>
Date: Thu, 27 Jul 2023 16:29:22 +0800
Subject: wpa_supplicant: Fix wpa_supplicant configuration parsing error

In the original flow, after hostapd_config_tx_queue successfully
parses a tx_queue variable, it would not return immediately. Then it
would print out "unknow global field" later and set return val to -1.

This patch returns immediately after hostapd_config_tx_queue
successfully parses a tx_queue variable.

Signed-off-by: Michael Lee <michael-cy.lee@mediatek.com>
---
 wpa_supplicant/config.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/wpa_supplicant/config.c b/wpa_supplicant/config.c
index bf062b0..de08fff 100644
--- a/wpa_supplicant/config.c
+++ b/wpa_supplicant/config.c
@@ -5397,6 +5397,7 @@ int wpa_config_process_global(struct wpa_config *config, char *pos, int line)
 					   line);
 				return -1;
 			}
+			return ret;
 		}
 
 		if (os_strncmp(pos, "wmm_ac_", 7) == 0) {
